sdk/samples/ping_app.c

```diff
@@ -0,0 +1,400 @@
+#define DEFAULT_SEND_IP "127.0.0.1"
+#define DEFAULT_SEND_PORT "10001"
+
+static volatile bool keepRunning = true;
+
+static MeshClient *client;
+atomic_int counter = 0;
+
+
+Config config = {
+    .recv_addr = DEFAULT_RECV_IP,
+    .recv_port = DEFAULT_RECV_PORT,
+    .send_addr = DEFAULT_SEND_IP,
+    .send_port = DEFAULT_SEND_PORT,
+    
+    .payload_type = "",
+    .protocol_type = "",
+    .pix_fmt_string = DEFAULT_VIDEO_FMT,
+    .socket_path = DEFAULT_MEMIF_SOCKET_PATH,
+    .interface_id = DEFAULT_MEMIF_INTERFACE_ID,
+    .loop = DEFAULT_INFINITE_LOOP,
+
+    .width = DEFAULT_FRAME_WIDTH,
+    .height = DEFAULT_FRAME_HEIGHT,
+    .vid_fps = DEFAULT_FPS,
+    .frame_size = 0,
+    .total_num = 300
+};
+
+void intHandler(int dummy)
+{
+    keepRunning = 0;
+}
+
+/* print a description of all supported options */
+void usage(FILE* fp, const char* path)
+{
+    /* take only the last portion of the path */
+    const char* basename = strrchr(path, '/');
+    basename = basename ? basename + 1 : path;
+
+    fprintf(fp, "usage: %s [OPTION]\n", basename);
+    fprintf(fp, "-H, --help\t\t\t"
+                "Print this help and exit\n");
+    fprintf(fp, "-w, --width=<frame_width>\t"
+                "Width of test video frame (default: %d)\n",
+        DEFAULT_FRAME_WIDTH);
+    fprintf(fp, "-h, --height=<frame_height>\t"
+                "Height of test video frame (default: %d)\n",
+        DEFAULT_FRAME_HEIGHT);
+    fprintf(fp, "-f, --fps=<video_fps>\t\t"
+                "Test video FPS (frame per second) (default: %0.2f)\n",
+        DEFAULT_FPS);
+    fprintf(fp, "-s, --ip=ip_address\t\t"
+                "Send data to IP address (default: %s)\n",
+        DEFAULT_SEND_IP);
+    fprintf(fp, "-p, --port=port_number\t\t"
+                "Send data to Port (default: %s)\n",
+        DEFAULT_SEND_PORT);
+    fprintf(fp, "-o, --protocol=protocol_type\t"
+                "Set protocol type (default: %s)\n",
+        DEFAULT_PROTOCOL);
+    fprintf(fp, "-n, --number=frame_number\t"
+                "Total frame number to send (default: %d)\n",
+        DEFAULT_TOTAL_NUM);
+    fprintf(fp, "-s, --socketpath=socket_path\t"
+                "Set memif socket path (default: %s)\n",
+        DEFAULT_MEMIF_SOCKET_PATH);
+    fprintf(fp, "-d, --interfaceid=interface_id\t"
+                "Set memif conn interface id (default: %d)\n",
+        DEFAULT_MEMIF_INTERFACE_ID);
+    fprintf(fp, "-l, --loop=is_loop\t"
+                "Set infinite loop sending (default: %d)\n",
+        DEFAULT_INFINITE_LOOP);
+    fprintf(fp, "-t, --threads=threads_num\t"
+                "Set threads_num (default: %d)\n",
+        1);
+    fprintf(fp, "\n");
+}
+
+typedef struct{
+    int thread_id;
+    double *latency_results;
+}t_data;
+
+void *receiver_thread(void *arg) {
+    t_data *thread_data = (t_data *)arg;
+    struct timespec send_time = {}, now = {};
+    int recved_pkt_num;
+    cpu_set_t cpuset;
+    int err;
+    int timeout = -1;
```

Comment:
Please rename timeout to timeout_ms for code readability

Reply:
done